src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt

```diff
@@ -233,7 +282,10 @@ private constructor(
     @RequiresEdt
     fun createFromState(project: Project, state: ChatState): AgentChatSession {
       val sessionId = createNewPanel(project) { it.server.chatNew() }
-      val chatSession = AgentChatSession(project, sessionId, state.internalId!!)
+      val selectedModel = state.model?.let { ChatModel.fromDisplayName(it) } ?: DEFAULT_MODEL
```

Comment:
Please at least ass a TODO there.
This might be incorrect in some scenarios, you should not assume what default model is.
Either fetch it from agent or change agent API to not require it in the first place.